node/cancellation-v2.md

```diff
@@ -0,0 +1,343 @@
+- `shield` - start the scope in `shield` mode preventing cancellation from propagating to inner scopes, Activities, timers, and Triggers.
+- `parent` - an **optional** `CancellationScope` (useful for running background tasks)
+
+#### `CancellationScope`
+
+- Passed into the function argument of `cancellationScope`
+
+  ```ts
+  await cancellationScope(async (scope /* CancellationScope */) => {
+    scope.deadline = Date.now() + 1000;
+    await httpPost('...');
+  });
+  ```
+
+- Exposes the following methods and attributes
+  - `cancelled` - a promise that rejects once the scope is cancelled
+  - `cancel()` - manually cancel the scope
+  - `deadline` - dynamic deadline for the scope
+  - `shield` - boolean controlling whether to shield scope from outer cancellation<br/>
+    (Scope may still be cancelled from within if it calls a function that throw `CancellationError`)
+
+#### `Trigger`
+
+- Implements the `PromiseLike` interface (awaitable via `then()`)
+- Scope aware: linked to parent scope and propagates cancellation
+- Provides public methods to `resolve` and `reject` itself
+- Useful for e.g waiting for a signal from WF main
+
+  ```ts
+  const unblocked = new Trigger<void>();
+  const signals = {
+    unblock(): void {
+      unblocked.resolve();
+    },
+  };
+
+  async function main() {
+    await unblocked;
+  }
+  ```
+
+## Examples
+
+#### Cancel a timer from Workflow code
+
+```ts
+import { CancellationError, cancellationScope, sleep } from '@temporalio/workflow';
+
+export async function main() {
+  // Timers and Activities are automatically cancelled when their containing scope is cancelled.
+  // Awaiting on a cancelled scope with throw the original CancellationError.
+  try {
+    await cancellationScope(async (scope) => {
+      sleep(1); // <-- Will be cancelled because it is attached to `scope`
+      scope.cancel();
+    });
+  } catch (e) {
+    if (e instanceof CancellationError) {
+      console.log('Exception was propagated 👍');
+    }
+    throw e; // <-- Fail the workflow
+  }
+}
+```
+
+#### Run multiple activities with a single deadline
+
+```ts
+import { withTimeout } from '@temporalio/workflow';
+import { httpGetJSON } from '@activities';
+
+export async function main(urls: string[], timeoutMs: number) {
+  // If timeout triggers before all activities complete
+  // the Workflow will fail with a CancellationError.
+  const results = await withTimeout(timeout, () => Promise.all(urls.map(httpGetJSON)));
+  // Do something with the results
+}
+```
+
+#### `CancellationScope`s can be shielded, preventing cancellation from propagating to created cancellables
+
+```ts
+import { cancellationScope } from '@temporalio/workflow';
+import { httpGetJSON } from '@activities';
+
+export async function main(url: string) {
+  // Prevent Activity from being cancelled and await completion.
+  // Note that the Workflow is completely oblivious to cancellation in this example.
```

Comment:
Oblivious, and also impervious, right? Like it just can't be cancelled effectively? Or is what's going to happen here is the activity won't be cancelled, but the workflow will be?

Reply:
Yes, impervious as well.